hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1192,7 +1195,7 @@ public StoreFileWriter createWriterInTmp(long maxKeyCount, Compression.Algorithm
     return builder.build();
   }
 
-  HFileContext createFileContext(Compression.Algorithm compression,
+  public HFileContext createFileContext(Compression.Algorithm compression,
```

Comment:
We have to make all this stuff public?

Reply:
This one in particular, could had logic duplicated wherever it's needed, but it's more of a helper method, so didn't think making it public would be harmful. I can revert it back to package private and re-implement this in the `DirectStoreCompactor`.